ocpp-v1_6/src/main/java/eu/chargetime/ocpp/wss/BaseWssSocketBuilder.java

```diff
@@ -93,22 +106,43 @@ public BaseWssSocketBuilder connectionTimeout(int connectionTimeout) {
 
     @Override
     public Socket build() throws IOException {
-        Socket socket = new Socket(proxy);
+        verify(true);
+
+        Socket socket = socketFactory.getSocket(proxy);
         socket.setTcpNoDelay(tcpNoDelay);
         socket.setReuseAddress(reuseAddr);
 
-        if( !socket.isBound() ) {
-            socket.connect(new InetSocketAddress(uri.getHost(), getPort(uri)), connectionTimeout);
+        if(!socket.isBound()) {
+            socket.connect(inetSocketAddressFactory.getInetSocketAddress(uri.getHost(), getPort(uri)),
+                    connectionTimeout);
         }
 
         return sslSocketFactory.createSocket(socket, uri.getHost(), getPort(uri), autoClose);
     }
 
     @Override
     public void verify() {
+        verify(false);
+    }
+
+    public interface SocketFactory {
+        Socket getSocket(Proxy proxy);
+    }
+
+    public interface InetSocketAddressFactory {
+        InetSocketAddress getInetSocketAddress(String host, int port);
+    }
+
+    private void verify(boolean complete) {
         if(sslSocketFactory == null) {
             throw new IllegalStateException("sslSocketFactory must be set");
         }
+
+        if(complete) {
```

Comment:
Put space before '(', please. There and in other places.

Reply:
I would propose to define reference for the project for a particular codestyle so it could be formatted automatically in accrodance (e.g. https://google.github.io/styleguide/javaguide.html)